### PR TITLE
chore: bump operator-rs to 0.60

### PR DESCRIPTION
**Pull request by fhennig**
# Description

see title

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] CRD documentation for all fields, following the [style guide](https://docs.stackable.tech/home/nightly/contributor/style-guide).
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
- [ ] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/style-guide).
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

### CODE REVIEW

**Comment by fhennig** [comment]
awaiting build to start a CI test.

File: rust/operator-binary/src/authentication/password/ldap.rs [hunk 1]
```diff
@@ -30,16 +22,26 @@ const LDAP_PASSWORD_ENV: &str = "LDAP_PASSWORD";
...
+
+    #[snafu(display("Failed to construct LDAP endpoint URL"))]
+    LdapEndpoint {
+        source: stackable_operator::commons::authentication::ldap::Error,
+    },
+
+    #[snafu(display("Failed to construct LDAP volumes and volume mounts"))]
+    LdapVolumesAndMounts {
+        source: stackable_operator::commons::authentication::ldap::Error,
+    },
```
Comment (maltesander) [inline rust/operator-binary/src/authentication/password/ldap.rs:34]: In the Kafka PR it was Volume and VolumeMounts (granted, another error). Should we maybe add something to the code guidelines to be consistent here?

Edit: its Volume and VolumeMounts in authentication/password/mod.rs as well.

File: rust/operator-binary/src/controller.rs [hunk 2]
```diff
@@ -1043,17 +1080,22 @@ fn build_rolegroup_statefulset(
...
-        .metadata_builder(|m| {
-            m.with_recommended_labels(build_recommended_labels(
-                trino,
-                &resolved_product_image.app_version_label,
-                &rolegroup_ref.role,
-                &rolegroup_ref.role_group,
-            ));
-            // This is actually used by some kuttl tests (as they don't specify the container explicitly)
-            m.with_annotation("kubectl.kubernetes.io/default-container", "trino")
-        })
```
Comment (maltesander) [inline rust/operator-binary/src/controller.rs:1056]: Can we keep the comment for the kuttl tests?

**Review by maltesander — CHANGES_REQUESTED** [review]

File: rust/operator-binary/src/controller.rs (hunk 2 above)
Comment (fhennig) [reply rust/operator-binary/src/controller.rs]: https://github.com/stackabletech/trino-operator/pull/522/commits/2fa075a939849106a01c9ce90f0010c980c20048

File: rust/operator-binary/src/authentication/password/ldap.rs (hunk 1 above)
Comment (fhennig) [reply rust/operator-binary/src/authentication/password/ldap.rs]: https://github.com/stackabletech/trino-operator/pull/522/commits/2a9368701cc44c9b268ea0f7dd7b464576995165

File: rust/operator-binary/src/authentication/password/file.rs [hunk 3]
```diff
@@ -5,7 +5,7 @@ use stackable_operator::{
     builder::{
         resources::ResourceRequirementsBuilder, ContainerBuilder, VolumeBuilder, VolumeMountBuilder,
     },
-    commons::authentication::StaticAuthenticationProvider,
+    commons::authentication::static_::AuthenticationProvider,
     commons::product_image_selection::ResolvedProductImage,
```
[the remark this answers is not in the document]
Comment (fhennig) [reply rust/operator-binary/src/authentication/password/file.rs]: https://github.com/stackabletech/trino-operator/pull/522/commits/77bd8215f2f5a7405928fdba177fb164788ffb1a

File: rust/operator-binary/src/controller.rs [hunk 4]
```diff
@@ -303,6 +305,32 @@ pub enum Error {
...
+
+    #[snafu(display("failed to build annotation"))]
+    AnnotationBuild {
+        source: stackable_operator::kvp::KeyValuePairError<Infallible>,
+    },
+
+    #[snafu(display("failed to build metadata"))]
+    MetadataBuild {
+        source: stackable_operator::builder::ObjectMetaBuilderError,
+    },
```
Comment (maltesander) [inline rust/operator-binary/src/controller.rs:328]:
```suggestion
    #[snafu(display("failed to get required Labels"))]
    GetRequiredLabels {
        source:
            stackable_operator::kvp::KeyValuePairError<stackable_operator::kvp::LabelValueError>,
    },

    #[snafu(display("failed to build Labels"))]
    LabelBuild {
        source: stackable_operator::kvp::LabelError,
    },

    #[snafu(display("failed to build Annotation"))]
    AnnotationBuild {
        source: stackable_operator::kvp::KeyValuePairError<Infallible>,
    },

    #[snafu(display("failed to build Metadata"))]
    MetadataBuild {
        source: stackable_operator::builder::ObjectMetaBuilderError,
    },
```

File: rust/operator-binary/src/authentication/password/ldap.rs [hunk 5]
```diff
@@ -30,16 +22,26 @@ const LDAP_PASSWORD_ENV: &str = "LDAP_PASSWORD";
...
+
+    #[snafu(display("Failed to construct LDAP endpoint URL"))]
+    LdapEndpoint {
+        source: stackable_operator::commons::authentication::ldap::Error,
+    },
+
+    #[snafu(display("Failed to construct LDAP volumes and volume mounts"))]
+    LdapVolumeAndVolumeMounts {
+        source: stackable_operator::commons::authentication::ldap::Error,
+    },
```
Comment (maltesander) [inline rust/operator-binary/src/authentication/password/ldap.rs:34]:
```suggestion
    #[snafu(display("Failed to construct LDAP Volumes and VolumeMounts"))]
    LdapVolumeAndVolumeMounts {
        source: stackable_operator::commons::authentication::ldap::Error,
    },
```

**Review by maltesander — APPROVED** [review]
LGTM!

**Pull request merged**